src/main/scala/com/microsoft/hyperspace/index/rules/JoinIndexRule.scala

```diff
@@ -50,7 +51,11 @@ import com.microsoft.hyperspace.index.rankers.JoinIndexRanker
  * These indexes are indexed by the join columns and can improve the query performance by
  * avoiding full shuffling of T1 and T2.
  */
-object JoinIndexRule extends Rule[LogicalPlan] with Logging {
+object JoinIndexRule
+    extends Rule[LogicalPlan]
+    with Logging
+    with HyperspaceEventLogging
+    with ActiveSparkSession {
```

Comment:
if you are using this, you must be removing some existing usage?

Reply:
Yes, thanks, my changes got reverted during a merge conflict fix. Fixed.